testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeColumnMappingMode.java

```diff
@@ -67,35 +67,245 @@ public void testColumnMappingModeName()
+                " 'delta.minReaderVersion'='2'," +
+                " 'delta.minWriterVersion'='5')");
+
+        try {
+            onDelta().executeQuery("" +
+                    "INSERT INTO default." + tableName + " VALUES " +
+                    "(1, array(struct('nested 1')), struct('databricks 1'),'ala'), " +
+                    "(2, array(struct('nested 2')), struct('databricks 2'), 'kota')");
+
+            List<Row> expectedRows = ImmutableList.of(
+                    row(1, "nested 1", "databricks 1", "ala"),
+                    row(2, "nested 2", "databricks 2", "kota"));
+
+            assertThat(onDelta().executeQuery("SELECT a_number, array_col[0].array_struct_element, nested.field1, a_string FROM default." + tableName))
+                    .containsOnly(expectedRows);
+            assertThat(onTrino().executeQuery("SELECT a_number, array_col[1].array_struct_element, nested.field1, a_string FROM delta.default." + tableName))
+                    .containsOnly(expectedRows);
+            assertThat(onTrino().executeQuery("SELECT a_string FROM delta.default." + tableName + " WHERE a_number = 1"))
+                    .containsOnly(ImmutableList.of(row("ala")));
+            assertThat(onTrino().executeQuery("SELECT a_number FROM delta.default." + tableName + " WHERE nested.field1 = 'databricks 1'"))
+                    .containsOnly(ImmutableList.of(row(1)));
+
+            // Verify the connector can read renamed columns correctly
+            onDelta().executeQuery("ALTER TABLE default." + tableName + " RENAME COLUMN a_number TO new_a_column");
+            onDelta().executeQuery("ALTER TABLE default." + tableName + " RENAME COLUMN nested.field1 TO field2");
+
+            assertThat(onTrino().executeQuery("DESCRIBE delta.default." + tableName))
+                    .containsOnly(ImmutableList.of(
+                            row("new_a_column", "integer", "", ""),
+                            row("array_col", "array(row(array_struct_element varchar))", "", ""),
+                            row("nested", "row(field2 varchar)", "", ""),
+                            row("a_string", "varchar", "", "")));
+
+            assertThat(onDelta().executeQuery("SELECT new_a_column, array_col[0].array_struct_element, nested.field2, a_string FROM default." + tableName))
+                    .containsOnly(expectedRows);
+            assertThat(onTrino().executeQuery("SELECT new_a_column, array_col[1].array_struct_element, nested.field2, a_string FROM delta.default." + tableName))
+                    .containsOnly(expectedRows);
+        }
+        finally {
+            onTrino().executeQuery("DROP TABLE delta.default." + tableName);
+        }
+    }
+
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, PROFILE_SPECIFIC_TESTS})
+    public void testColumnMappingModeNameWithNonLowerCaseColumnName()
+    {
+        String tableName = "test_dl_column_mapping_mode_name_" + randomTableSuffix();
```

Comment:
Test above uses "test_dl_column_mapping_mode_name", this one adds an underscore
let's be more unique.
(i know table names are randomized, but unique prefixes allow you to track which test created a particular table)